packages/vite/src/node/plugins/resolve.ts

```diff
@@ -1117,35 +1109,31 @@ function packageEntryFailure(id: string, details?: string) {
 function resolveExportsOrImports(
   pkg: PackageData['data'],
   key: string,
-  options: InternalResolveOptionsWithOverrideConditions,
+  options: InternalResolveOptions,
   type: 'imports' | 'exports',
 ) {
-  const additionalConditions = new Set(
-    options.overrideConditions || [
-      'production',
-      'development',
-      'module',
-      ...options.conditions,
-    ],
+  const conditions = [...options.conditions, 'require', 'import'].filter(
```

Comment:
I didn't give users a way to remove `require` / `import` as I didn't understand how that would work.